lib/web_ui/lib/src/engine/view_embedder/flutter_view_manager.dart

```diff
@@ -101,5 +110,49 @@ class FlutterViewManager {
     // close the streams.
     _onViewCreatedController.close();
     _onViewDisposedController.close();
+
+    // Remove the focus tracking event listeners.
+    domDocument.body?.removeEventListener(_focusin, _focusChangeHandler, true);
+    domDocument.body?.removeEventListener(_focusout, _focusChangeHandler, true);
   }
+
+  int? _latestFocusedViewId;
+
+  // TODO(tugorez): Add handlers for backwards focus.
+  late final DomEventListener _focusChangeHandler = createDomEventListener((DomEvent event) {
```

Comment:
Is it a good idea to add focus management to this existing class? Or should we create a separate class for managing view focus? I'm also curious what @ditman thinks about this.

Reply:
IMO `view focus` is something that should be handled here or in an entity higher in the hierarchy. Happy to create a helper class that abstract those behaviors tho. Logic seemed simple so decided not to proceed with that approach. This is an internal implementation so should be straightforward to change in the future.

Reply:
Thought about creating a `view focus manager` but feels weird having that one and the `view manager`. Open to suggestions :)

Reply:
The current `view manager` is more like a `view registry`. It keeps track of all existing views and handles registration/unregistration. That's very different from focus management. That's just my opinion though, and I don't feel very strongly about it. I'll leave it up to you and David (the original author of this class).

Reply:
Makes sense. Done